initial solution

MarcusPy opened the pull request:
In ColorSupplier class I decided to make field colors a static final because in my research, I found out that it will be more performance friendly this way, considering we want to fetch values from the Enum only once throughout the entire lifetime of the program. I also made the method getRandomColor static because we use it as a utility, so creating an instance seems redundant here. I hope the choices I made are right

nowickimj commented on src/main/java/core/basesyntax/ColorSupplier.java:
```diff
@@ -1,7 +1,14 @@
+    private static final Color[] colors = Color.values();
+
+    public static Color getRandomColor() {
+        Random random = new Random();
```
Make this a constant.

nowickimj commented on src/main/java/core/basesyntax/Lottery.java:
```diff
@@ -0,0 +1,13 @@
+public class Lottery {
+    public Ball getRandomBall() {
+        Color color = ColorSupplier.getRandomColor();
+        Random random = new Random();
```
this can be a constant as well.

nowickimj commented on src/main/java/core/basesyntax/ColorSupplier.java:
```diff
@@ -1,7 +1,14 @@
-        return null;
+    private static final Color[] colors = Color.values();
+
+    public static Color getRandomColor() {
```
This method shouldn't be static

MarcusPy commented:
Addressed the issues You've highlighted, although is storing an instance of Random as a constant a good practice? I've read that it is more performance friendly, but also reduces the randomness of it due to everything being generated from a single seed

okuzan commented on src/main/java/core/basesyntax/Application.java:
```diff
@@ -2,6 +2,10 @@
-        // create three balls using class Lottery and print information about them in console
+        Lottery lottery = new Lottery();
+        for (int i = 0; i < 3; i++) {
+            Ball ball = lottery.getRandomBall();
+            System.out.println(ball.toString());
```
```suggestion
            System.out.println(lottery.getRandomBall());
```
toString is called under the hood on the argument of println

okuzan commented on src/main/java/core/basesyntax/Lottery.java:
```diff
@@ -0,0 +1,14 @@
+
+    public Ball getRandomBall() {
+        Color color = new ColorSupplier().getRandomColor();
+        int randomNumber = random.nextInt(101);
```
101 should be a class level constant

nowickimj reviewed (approved):
I will approve this PR because this is the final iteration, but one small remark: when creating constants, please use UPPER_CASE for names.

Regarding using Random instance as a constant - that's very good catch, although we may focus on clean code practices for now. There are better ways for proper randomization :)
